pkg/expressions/expressions.go

```diff
@@ -334,3 +334,24 @@ func (e *expressionEvaluator) parseAttributes(ctx context.Context) error {
 	e.attrs = attrs
 	return nil
 }
+
+type CompileError struct {
+	Err error
+}
+
+func NewCompileError(err error) *CompileError {
```

Comment:
I've introduced a custom error to

- easily check for the occurrence in executor.go
- provide the underlying expression compilation error to the user without including the intermediate error messages